src/openforms/registrations/contrib/objects_api/templatetags/registrations/contrib/objects_api/json_tags.py

```diff
@@ -1,27 +1,33 @@
 import json
+import warnings
 
 from django import template
 from django.conf import settings
 from django.utils.safestring import SafeString
 
+from glom import Glommer
+
 from openforms.formio.rendering.structured import render_json
 from openforms.registrations.contrib.objects_api.utils import html_escape_json
+from openforms.typing import JSONObject, JSONValue
 
 register = template.Library()
 
 
 @register.simple_tag(takes_context=True)
-def uploaded_attachment_urls(context):
+def uploaded_attachment_urls(context: template.Context) -> str:
     """
     Output a sequence of attachment URLs as a JSON-serialized list.
     """
-    # attachments is a list of URLs
-    attachments = context.get("submission", {}).get("uploaded_attachment_urls", [])
-    return SafeString(json.dumps(attachments))
+    warnings.warn(
+        "'uploaded_attachment_urls' is deprecated, use 'as_json' with a path argument instead.",
+        DeprecationWarning,
+    )
+    return as_json(context, "submission.uploaded_attachment_urls")
 
 
 @register.simple_tag(takes_context=True)
-def json_summary(context):
+def json_summary(context: template.Context) -> str:
```

Comment:
type hint is wrong, it returns a `SafeString`, not just a bare `str`. And the if-branch on line 32 result in a dict being returned.